test/jdk/javax/net/ssl/HttpsURLConnection/SubjectAltNameIPv6.java

```diff
@@ -0,0 +1,361 @@
+    volatile Exception serverException = null;
+
+    // SSLSocket object created by HttpsClient internally
+    SSLSocket clientSSLSocket = null;
+
+    // the hostname the server socket is bound to
+    String hostName;
+
+    /*
+     * Define the server side of the test.
+     *
+     * If the server prematurely exits, serverReady will be set to true
+     * to avoid infinite hangs.
+     */
+    void doServerSide() throws Exception {
+        SSLServerSocketFactory sslssf =
+            new SimpleSSLContext().get().getServerSocketFactory();
+        SSLServerSocket sslServerSocket =
+            (SSLServerSocket) sslssf.createServerSocket(
+                    serverPort, 0,
+                    InetAddress.getByName(hostName));
+        sslServerSocket.setEnabledProtocols(new String[]{"TLSv1.3"});
+        serverPort = sslServerSocket.getLocalPort();
+
+        /*
+         * Signal the client, the server is ready to accept connection.
+         */
+        serverReady.countDown();
+
+        SSLSocket sslSocket = (SSLSocket) sslServerSocket.accept();
+        OutputStream sslOS = sslSocket.getOutputStream();
+        BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(sslOS));
+        bw.write("HTTP/1.1 200 OK\r\n\r\n\r\n");
+        bw.flush();
+        sslSocket.close();
+    }
+
+    /*
+     * Define the client side of the test.
+     *
+     * If the server prematurely exits, serverReady will be set to true
+     * to avoid infinite hangs.
+     */
+    void doClientSide() throws Exception {
+
+        /*
+         * Wait for server to get started.
+         */
+        boolean ready = serverReady.await(10, TimeUnit.SECONDS);
+        if (!ready) {
+            throw new RuntimeException("Server timed out.");
+        }
+        if (serverException != null) {
+            throw new RuntimeException("Server failed to start.", serverException);
+        }
+
+        SSLSocketFactory sf = new SimpleSSLContext().get().getSocketFactory();
+        URI uri = new URI("https://" + hostName + ":" + serverPort + "/index.html");
+        HttpsURLConnection conn = (HttpsURLConnection)uri.toURL().openConnection();
+
+        /*
+         * Simulate an external JSSE implementation and store the client SSLSocket
+         * used internally.
+         */
+        conn.setSSLSocketFactory(wrapSocketFactory(sf,
+                sslSocket -> clientSSLSocket = sslSocket));
+        conn.getInputStream();
+
+        var sniSN = clientSSLSocket.getSSLParameters().getServerNames();
+        if( sniSN != null && !sniSN.isEmpty()) {
+            throw new RuntimeException("SNI server name '" +
+                    sniSN.getFirst() + "' must not be set.");
+        }
+
+        if (conn.getResponseCode() == -1) {
+            throw new RuntimeException("getResponseCode() returns -1");
+        }
+    }
+
+    public static void main(String[] args) throws Exception {
+
+        if (!IPSupport.hasIPv6()) {
```

Comment:
```suggestion
        if (IPAddressUtil.isIPv6LiteralAddress(args[0]) && !IPSupport.hasIPv6()) {
```

Note that you need to import `sun.net.util.IPAddressUtil` first.